### PR TITLE
Add specializations for vectorized conversion between float and BF16/FP16

### PR DESCRIPTION
**Pull request by CaoE**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #133830
* #126502
* #126501
* __->__ #126500

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by jgong5 — CHANGES_REQUESTED** [review]
Please fix CI failures...

**Review by jgong5 — APPROVED** [review]

File: torch/_inductor/codegen/cpp_prefix.h [hunk 1]
```diff
@@ -199,6 +203,23 @@ Welford<scalar_t> welford_vec_reduce_all(Welford<at::vec::Vectorized<scalar_t>>
...
+    acc.weight[0]
+  };
+  auto Welford1 = Welford<at::vec::Vectorized<scalar_t>>{
+    acc.mean[1],
+    acc.m2[1],
+    acc.weight[1]
+  };
+  auto result0 = welford_vec_reduce_all(Welford0);
+  auto result1 = welford_vec_reduce_all(Welford1);
+  return welford_combine(result0, result1);
```
[the remark this answers is not in the document]
Comment (CaoE) [reply torch/_inductor/codegen/cpp_prefix.h]: Thanks for your comments. This change has been merged in another PR.

**Comment by CaoE** [comment]
@peterbell10 Could you please review this PR ? Thank you.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**